app/src/firebase/snaps/SnapService.ts

```diff
@@ -57,3 +57,22 @@ export function streamSubmittedSnapsForCurrentUser(
             },
         });
 }
+
+export function streamNumberOfSnapsForSnappable(
+    onNumberReceived: (numSnaps: number) => void,
+    onError: (error: Error) => void,
+    snappableId: String
+): () => void {
+    return firebase
+        .firestore()
+        .collection("snaps")
+        .where("to", "array-contains", snappableId)
+        .onSnapshot({
+            next: (querySnapshot) => {
+                onNumberReceived(querySnapshot.docs.length);
+            },
+            error: (error) => {
+                onError(error);
+            },
+        });
+}
```

Comment:
This will have to be updated to run on the current cup (sorry)! Adam and Emily's changes there are merged though, so that should be relatively straight forward.

I also think this exact approach may suffer from a performance penalty (i.e. is it getting up a separate stream for each snappable?).

It would be better to stream the snaps in the current cup and then transform that in the browser by the `to` field. That should be less network heavy and a bit cleaner overall.